File: application/src/main/java/de/pbauerochse/worklogviewer/fx/converter/GroupingComboBoxConverter.java
package de.pbauerochse.worklogviewer.fx.converter;

import de.pbauerochse.worklogviewer.util.FormattingUtil;
import de.pbauerochse.worklogviewer.view.grouping.Grouping;
import javafx.scene.control.ComboBox;
import javafx.util.StringConverter;
import org.apache.commons.lang3.StringUtils;
import org.jetbrains.annotations.NotNull;

import java.util.Optional;

public class GroupingComboBoxConverter extends StringConverter<Grouping> {

    private final ComboBox<Grouping> categoryComboBox;

    public GroupingComboBoxConverter(@NotNull ComboBox<Grouping> categoryComboBox) {
        this.categoryComboBox = categoryComboBox;
    }

    @Override
    public String toString(Grouping category) {
        return Optional.ofNullable(category)
                .map(Grouping::getLabel)
                .orElseGet(() -> FormattingUtil.getFormatted("grouping.none"));
    }

    @Override
    public Grouping fromString(String categoryName) {
        // special "nothing-selected" item
        if (StringUtils.equals(FormattingUtil.getFormatted("grouping.none"), categoryName)) {
            return null;
        }

        return categoryComboBox.getItems().stream()
                .filter(category -> StringUtils.equals(category.getLabel(), categoryName))
                .findFirst()
                .orElse(null);
    }
}

File: application/src/main/java/de/pbauerochse/worklogviewer/fx/converter/GroupingComboBoxConverter.kt
package de.pbauerochse.worklogviewer.fx.converter

import de.pbauerochse.worklogviewer.util.FormattingUtil.getFormatted
import de.pbauerochse.worklogviewer.view.grouping.Grouping
import javafx.scene.control.ComboBox
import javafx.util.StringConverter

class GroupingComboBoxConverter(private val categoryComboBox: ComboBox<Grouping>) : StringConverter<Grouping>() {

    override fun toString(category: Grouping?): String {
        return category?.label ?: getFormatted("grouping.none")
    }

    override fun fromString(categoryName: String?): Grouping? {
        // special "nothing-selected" item
        return if (getFormatted("grouping.none") == categoryName) {
            null
        } else categoryComboBox.items.firstOrNull { it.label == categoryName }
    }
}
